src/agentscope/rpc/rpc_agent_client.py

```diff
@@ -34,63 +46,287 @@ def __init__(self, host: str, port: int, agent_id: str = "") -> None:
+            Defaults to 300.
 
         Returns:
             str: serialized return data.
         """
-        with grpc.insecure_channel(f"{self.host}:{self.port}") as channel:
-            stub = RpcAgentStub(channel)
-            result_msg = stub.call_func(
-                RpcMsg(
-                    value=value,
-                    target_func=func_name,
-                    agent_id=self.agent_id,
-                ),
-                timeout=timeout,
-            )
-            return result_msg.value
+        try:
+            with grpc.insecure_channel(f"{self.host}:{self.port}") as channel:
+                stub = RpcAgentStub(channel)
+                result_msg = stub.call_agent_func(
+                    agent_pb2.RpcMsg(
+                        value=value,
+                        target_func=func_name,
+                        agent_id=self.agent_id,
+                    ),
+                    timeout=timeout,
+                )
+                return result_msg.value
+        except Exception as e:
+            # check the server and raise a more reasonable error
+            if not self.is_alive():
+                raise AgentServerNotAliveError(
+                    host=self.host,
+                    port=self.port,
+                    message=str(e),
+                ) from e
+            raise e
+
+    def is_alive(self) -> bool:
+        """Check if the agent server is alive.
+
+        Returns:
+            bool: indecates whether the server is alive.
+        """
 
-    def create_agent(self, agent_configs: dict) -> None:
-        """Create a new agent for this client."""
         try:
-            if self.agent_id is None or len(self.agent_id) == 0:
-                return
-            self.call_func(
-                "_create_agent",
-                base64.b64encode(dill.dumps(agent_configs)).decode("utf-8"),
+            with grpc.insecure_channel(f"{self.host}:{self.port}") as channel:
+                stub = RpcAgentStub(channel)
+                status = stub.is_alive(Empty(), timeout=5)
+                if not status.ok:
+                    raise AgentServerNotAliveError(
+                        host=self.host,
+                        port=self.port,
+                    )
+                return status.ok
+        except Exception:
+            logger.info(
+                f"Agent server [{self.host}:{self.port}] not alive.",
             )
+            return False
+
+    def stop(self) -> None:
+        """Stop the agent server."""
+        try:
+            with grpc.insecure_channel(f"{self.host}:{self.port}") as channel:
+                stub = RpcAgentStub(channel)
+                logger.info(
+                    f"Stopping agent server at [{self.host}:{self.port}].",
+                )
+                resp = stub.stop(Empty(), timeout=5)
+                if resp.ok:
+                    logger.info(
+                        f"Agent server at [{self.host}:{self.port}] stopped.",
```

Comment:
Considering the logging in lines 123-125, maybe f"Finished" is enough?

Reply:
For debugging, used to observe the situation where some of the servers try to stop but failed to stop successfully.